File: 0258-add-digits/0258-add-digits.java
class Solution {
    
    public int addDigits(int num) {
        return sumDigits(num);    
    }

    int sumDigits(int num) {
       if(num <= 9) {
           return num;
       }
        
        num = num%10 + sumDigits(num/10);
        
        return sumDigits(num);
    }
    
}

File: 0258-add-digits/0258-add-digits.kt
class Solution {
    fun addDigits(num: Int): Int {
            var sum = num.toString()
    while (sum.length != 1){
        sum = sum.split("")
            .filter { it != "" }
            .sumBy { it.toInt() }
            .toString()
    }
    return sum.toInt()
    }
}